sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -473,13 +473,13 @@ public static UnsafeArrayData fromPrimitiveArray(
     return result;
   }
 
-  public static UnsafeArrayData forPrimitiveArray(int offset, int length, int elementSize) {
-    return fromPrimitiveArray(null, offset, length, elementSize);
+  public static UnsafeArrayData forPrimitiveArray(int length, int elementSize) {
+    return fromPrimitiveArray(null, 0, length, elementSize);
```

Comment:
is it safe? I vaguely remember the address of offheap memory block is usually not 0.

Reply:
so this is used to create a new unsafe array with no data. This saves duplicated code, but we will do unnecessary memory copy.

Shall we just add a new method in `UnsafeArrayData` to create a fresh array?